provisioner/clusterpy.go

```diff
@@ -625,6 +630,21 @@ func (p *clusterpyProvisioner) Decommission(ctx context.Context, logger *log.Ent
 		logger.Errorf("Unable to downscale the deployments, proceeding anyway: %s", err)
 	}
 
+	// decommission each node pool
+	// TODO: for now only karpenter pools are supported. Other pools are
+	// decommissioned by the stack deletion logic below.
+	// Would be cleaner to have a per node pool decommissioning logic.
+	for _, nodePool := range cluster.NodePools {
+		if nodePool.Profile == karpenterNodePoolProfile {
+			logger.Infof("Decommissioning Node Pool: %s (profile: %s)", nodePool.Name, nodePool.Profile)
+			nodePoolBackend := updatestrategy.NewEC2NodePoolBackend(cluster.ID, awsAdapter.session)
+			err := nodePoolBackend.Decommission(ctx, nodePool)
```

Comment:
According to the docs the recommended approach to decommission karpenter nodes is to delete the `Node` object via the Kubernetes API. Worked fine when I tried it.

Reply:
Actually, deleting the provisioner resource should be enough, and karpenter shall delete the nodes automatically.
I see your point that terminating the nodes from the AWS API is not needed.

but what do you think
1. only delete the provisioner resource and leave it to karpenter to clean up the nodes
2. delete the provisioner resource and delete the node objects from kubectl
3. delete the provisioner resource and label the nodes for cluster-lifecycle-manager to decommission

Reply:
I prefer 1. over 2.

Regarding 3: We definitely need to check compatibility with cluster-lifecycle-manager (CLC). We have some logic there (or in CA) that's specific to Zalando (e.g. the forced eviction of Pods after 3 days). Ideally karpenter's way of cleaning up (which respects PDBs already) should be the only thing we need (maybe we need to contribute back if there's something missing). 

Otherwise we need to somehow make it work with CLC.

Reply:
I will go with 1 for now, until we do better discovery

Reply:
actually, there are 2 different scenarios where we need to decommission a node pool

-a) configuring the cluster with `zregistry` deleting a node pool
-b) decommissioning the whole cluster

what we discussed fits more for the first scenario, for the later we need to actively delete the nodes in order not to wait until karpenter controller cleans them

Reply:
We can also leave it like it is for now if it works.

Reply:
Alternatively, as part of the cluster decommission, we can start by deleting the karpenter provisioners and wait until karpenter controller delete all nodes.

then we downscale kube-system controllers
this is a bit different behavior than what we have 2day